packages/node/src/transports/new.ts

```diff
@@ -0,0 +1,151 @@
+
+interface HttpTransportOptions extends BaseTransportOptions {
+  // Todo: doc
+  headers?: Record<string, string>;
+  // TODO: doc Set a HTTP proxy that should be used for outbound requests.
+  proxy?: string;
+  // TODO: doc HTTPS proxy certificates path
+  caCerts?: string;
+  // Todo: doc
+  httpModule?: HTTPModule;
+}
+
+// TODO(v7):
+// - Rename this file "transports.ts"
+// - Move this file one folder upwards
+// - Delete "transports" folder
+
+/**
+ * TODO Doc
+ */
+export function makeNewHttpTransport(options: HttpTransportOptions): NewTransport {
+  // Proxy prioritization: http  => `options.proxy` | `process.env.http_proxy`
+  const proxy = filterNoProxy(options.url, options.proxy || process.env.http_proxy);
+
+  const httpModule = options.httpModule ?? http;
+
+  const agent = proxy
+    ? (new (require('https-proxy-agent'))(proxy) as http.Agent)
+    : new http.Agent({ keepAlive: false, maxSockets: 30, timeout: 2000 });
+
+  const requestExecutor = createRequestExecutor(options, httpModule, agent);
+  return createTransport({ bufferSize: options.bufferSize }, requestExecutor);
+}
+
+/**
+ * TODO Doc
+ */
+export function makeNewHttpsTransport(options: HttpTransportOptions): NewTransport {
+  // Proxy prioritization: https => `options.proxy` | `process.env.https_proxy` | `process.env.http_proxy`
+  const proxy = filterNoProxy(options.url, options.proxy || process.env.https_proxy || process.env.http_proxy);
+
+  const httpsModule = options.httpModule ?? https;
+
+  const agent = proxy
+    ? (new (require('https-proxy-agent'))(proxy) as https.Agent)
+    : new https.Agent({ keepAlive: false, maxSockets: 30, timeout: 2000 });
+
+  const requestExecutor = createRequestExecutor(options, httpsModule, agent);
+  return createTransport({ bufferSize: options.bufferSize }, requestExecutor);
+}
+
+/**
+ * Honors the `no_proxy` env variable with the highest priority to allow for hosts exclusion.
+ *
+ * @param transportUrl The URL the transport intends to send events to.
+ * @param proxy The client configured proxy.
+ * @returns A proxy the transport should use.
+ */
+function filterNoProxy(transportUrl: string, proxy: string | undefined): string | undefined {
+  const { no_proxy } = process.env;
+
+  const urlIsExemptFromProxy = no_proxy && no_proxy.split(',').some(exemption => transportUrl.endsWith(exemption));
+
+  if (urlIsExemptFromProxy) {
+    return undefined;
+  } else {
+    return proxy;
+  }
+}
+
+/**
+ * TODO Doc
+ */
+function createRequestExecutor(
+  options: HttpTransportOptions,
+  httpModule: HTTPModule,
+  agent: http.Agent,
+): TransportRequestExecutor {
+  const { hostname, pathname, port, protocol } = new URL(options.url);
+
+  // This function is extracted because we want to keep the actual `makeRequest` function as light-weight as possible
+  function performHttpRequest(
+    callback: (transportMakeRequestResponse: TransportMakeRequestResponse) => void,
```

Comment:
Tbh I don't think extracting into a function makes a whole lot of difference, especially since it's still in the same closure, so it's not like we extracted to make unit testing easier. Would rather just have `const req = httpModule.request` below.

Reply:
Yeah that seems fair. I changed it: 0d77e8062a83be1c2451a4894d8a6afb348ce5f7